### PR TITLE
first attempt at testing Module and ModuleCollection class

### PR DESCRIPTION
**Pull request by shahzebsiddiqui**

### CODE REVIEW

File: tests/test_modules.py [hunk 1]
```diff
@@ -60,3 +60,59 @@ def test_module_diff_invalid_args():
...
+
+    @pytest.mark.xfail(
+        reason="Collection Name must be string when saving", raises=TypeError
+    )
+    def test_type_error_save_collection(self):
+        cmd = Module(["settarg"])
+        cmd.save(1)
+
+    @pytest.mark.xfail(
+        reason="Collection Name must be string when showing content of collection",
```
Comment (vsoch) [inline tests/test_modules.py:96]: I think I had some comments about this in the previously closed PR - and of course I can't find the link now!

**Comment by vsoch** [comment]
Make sure that the version of black you have locally matches the linter on the action:

```
black-19.10b0-py_0
```
or locally
```
pip install black==19.10b
```

File: tests/test_modules.py (hunk 1 above)
Comment (vsoch) [reply tests/test_modules.py:96]: here it is! Your comment led to me to it :) https://github.com/HPC-buildtest/buildtest-framework/pull/185

**Comment by shahzebsiddiqui** [comment]
I am going to merge this since it increased coverage by **2.21%**. I will refactor the test, by first refactoring the ``ModuleCollection`` class defined in #188

**Pull request merged**